components/Sidebar/utils.ts

```diff
@@ -42,10 +38,25 @@ export const sortByLatest = (a, b) => {
   );
 };
 
-export const sortingFunctions = {
-  [ESortKeys.Latest]: sortByLatest,
-  [ESortKeys.Oldest]: sortByOldest,
-  [ESortKeys.Distance]: sortByDistance
+export const sortingFunctions = (sortKey, sortDirection) => {
+  switch (sortKey) {
+    case ESortKeys.Date: {
+      switch (sortDirection) {
+        case ESortDirection.Ascending:
+          return sortByLatest;
+        case ESortDirection.Descending:
+          return sortByOldest;
+      }
+    }
+    case ESortKeys.Distance: {
+      switch (sortDirection) {
+        case ESortDirection.Ascending:
+          return sortByDistanceAscending;
+        case ESortDirection.Descending:
+          return sortByDistanceDescending;
+      }
+    }
+  }
```

Comment:
```suggestion
export const sortingFunctions = (sortKey, sortDirection) => {
  switch (sortKey) {
    case ESortKeys.Date: {
    	return sortByDate(sortDirection);
    }
    case ESortKeys.Distance: {
    	return sortByDistance(sortDirection);
    }
  }
```

Since `sortByOldest` and `sortByLatest` do the same, I think creating only one function with a comparison inside (`sortDirection` being the comparison term) would make more sense (same for `sortByDistanceAscending` and `sortByDistanceDescending`).